packages/grafana-data/src/utils/labels.ts

```diff
@@ -48,6 +48,12 @@ export function findCommonLabels(labelsSets: Labels[]): Labels {
   }, (undefined as unknown) as Labels);
 }
 
+/** replace labels in a string.  Used for loki+prometheus legend formats */
+export function renderLabelsTemplate(aliasPattern: string, aliasData: Labels): string {
+  const aliasRegex = /\{\{\s*(.+?)\s*\}\}/g;
```

Comment:
maybe?

```suggestion
const aliasRegex = /\{\{\s*([^{}]+?)\s*\}\}/g;
```

Reply:
but probably this, which is resilient to bad syntax. this one also assumes there cannot be spaces in the variable name itself.

```suggestion
  const aliasRegex = /(?:\{+\s*)([^{}\s]+)(?:\s*\}+)/g;
```

Reply:
Not sure if we need/want to support the these variations, but the pattern above fails for:
![image](https://user-images.githubusercontent.com/705951/145874123-51f55e73-40fd-412f-8c0e-84a7928ddbea.png)